deku-derive/src/macros/deku_read.rs

```diff
@@ -220,6 +225,13 @@ fn emit_enum(input: &DekuData) -> Result<TokenStream, syn::Error> {
             let initialize_enum =
                 super::gen_enum_init(variant_is_named, variant_ident, internal_fields);
 
+            if let Some(deku_id_deref) = deku_id_deref {
+                let variant_deku_id = variant.id.as_ref();
+                let deku_id =
+                    quote! { Self :: #initialize_enum => Ok(#deku_id_deref #variant_deku_id)};
+                deku_ids.push(deku_id);
+            }
+
```

Comment:
It may make more sense to do the deref logic here:

```suggestion
            if let Some(variant_id) = &variant.id {
                let deref = match variant_id {
                    Id::TokenStream(v) => quote!{},
                    Id::LitByteStr(v) => quote! {*},
                };
                let deku_id =
                    quote! { Self :: #initialize_enum => Ok(#deref #variant_id)};
                deku_ids.push(deku_id);
            }

```

Reply:
Fixed